dts/bindings/usb-c/usb-c-connector.yaml

```diff
@@ -0,0 +1,173 @@
+# Copyright 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    A USB Type-C connector node represents a physical USB Type-C connector.
+    It should be a child of a USB-C interface controller or a separate node
+    when it is attached to both MUX and USB-C interface controller.
+
+    This is based on Linux, documentation:
+      https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/connector/usb-connector.yaml?h=v5.19&id=3d7cb6b04c3f3115719235cc6866b10326de34cd
+
+compatible: "usb-c-connector"
+
+include: base.yaml
+
+properties:
+    tcpc:
+        type: phandle
+        required: true
+        description: |
+            Type-C Port Controller for this port.
+
+    vbus:
+        type: phandle
+        required: true
+        description: |
+            VBUS measurement and control for this port.
+
+    power-role:
+        type: string
+        required: true
+        enum:
+          - "SINK"
+          - "SOURCE"
+          - "DUAL"
+        description: |
+          The Port power role. "DUAL" for Dual Role Port.
+
+    try-power-role:
+        type: string
+        required: false
+        enum:
+          - "SINK"
+          - "SOURCE"
+          - "DUAL"
+        description: |
+          Preferred power role.
+
+    data-role:
+        type: string
+        required: false
+        enum:
+          - "UFP"
```

Comment:
can these not match the linux binding of `host`, `device`, and `dual`?

Reply:
The spec and all the macros use UFP an DFP. I thought it would be clearer if we used UFP and DFP instead of Device and Host.

Reply:
we should be trying to match the linux binding as much as possible.  So I think its better to use `host` and `device` and just add a comment in the binding about those mapping to `UFP` and `DFP`

Reply:
Done